agent/proto/agent.proto

```diff
@@ -42,7 +43,13 @@ message WorkspaceApp {
 	}
 	Health health = 12;
 	bool hidden = 13;
-	string open_in = 14;
+
+	enum OpenIn {
+		WINDOW = 0;
+		SLIM_WINDOW = 1;
+		TAB = 2;
+	}
+	OpenIn open_in = 14;
```

Comment:
This change is not backwards-compatible; a string cannot be coerced to an enum/varint.

```suggestion
        string open_in = 14 [deprecated = true];

	enum OpenInMode {
		WINDOW = 0;
		SLIM_WINDOW = 1;
		TAB = 2;
	}
	OpenInMode open_in_mode = 15;
```